### PR TITLE
ESlint 9

### PR DESCRIPTION
**Pull request by jaap3**
- [x] Convert vue2 config
- [x] Convert vue3 config
- [ ] ~~Convert nuxt2 config~~
- [ ] ~~Convert nuxt3 config~~
- [x] Update package-lock.json
- [x] Update configuration to mention each config
- [x] Add automated tests
- [x] Test configs in AT project

Dropped Nuxt 2 support (not possible anymore, use the vue2 config for that if necessary).

Nuxt 3 has a module that configures eslint based on the project settings, which is more powerful than we can do with this package. Added instructions on how to integrate our overrides with the nuxt generated config.

Tested everything in the AT frontend monorepo, which has plain javascript, vue2, typescript and vue 3/nuxt subprojects.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,105 @@
...
 
-If you already had Eslint installed in your project you can delete this first with `npm uninstall eslint`. There may also be other Eslint packages you want to remove.
+If you already had ESlint installed in your project you can delete this first with `npm uninstall eslint`.
+There may also be other ESlint packages (plugins and configs) you want to remove.
 
 Then install this package as a development dependency `npm i -D @leukeleu/eslint-config`.
 
 ## Adding the config
 
 In the configs folder you can find all the Eslint config files to choose from.
```
Comment (dbunskoek) [inline README.md:14]: Eslint -> ESlint

File: README.md [hunk 2]
```diff
@@ -1,48 +1,105 @@
...
-  "lint:js": "eslint . --ext .vue,.js,.ts --cache",
-  "lintfix:js": "npm run lint:js -- --fix",
+import eslintConfig from '@leukeleu/eslint-config/configs/vue3'
+
+export default eslintConfig
+```
+
+### Nuxt 3
+
+Nuxt has its own ESLint module that configures ESLint based on the Nuxt project configuration. This generates a config
```
Comment (dbunskoek) [inline README.md:51]: ESLint -> ESlint (multiple times)

File: README.md [hunk 3]
```diff
@@ -1,48 +1,105 @@
...
+file that knows about the Nuxt project structure, layers, auto-imports, etc.
```
Comment (dbunskoek) [inline README.md:52]: maybe mention the name / location of that config file here, so you more easily recognize it

File: README.md (hunk 2 above)
Comment (jaap3) [reply README.md:51]: ESLint is correct (https://www.eslint.org)

File: README.md [hunk 4]
```diff
@@ -1,48 +1,105 @@
...
+  "scripts": {
+    "lint:js": "eslint --cache",
+    "lintfix:js": "npm run lint:js -- --fix"
+  }
 }
 ```
 
 ## Running the check
 
 Now you can run the Eslint check using `npm run lint:js` or `npm run lintfix:js` to auto fix.
```
Comment (dbunskoek) [inline README.md:82]: Eslint -> ESlint

File: configs/base.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
+/**
+ * ESLint flat configurations for Leukeleu.
```
Comment (dbunskoek) [inline configs/base.js:2]: ESLint -> ESlint

File: configs/base.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ * These configs must be combined for a full ESLint configuration.
```
Comment (dbunskoek) [inline configs/base.js:4]: idem

**Review by dbunskoek — APPROVED** [review]
really nice work, love the testing approach

only remarks that I had are about consistently naming ESlint, not sure if we want to 'pull this straight' now

**Comment by jaap3** [comment]
> really nice work, love the testing approach
> 
> only remarks that I had are about consistently naming ESlint, not sure if we want to 'pull this straight' now

I believe I now consistently call it ESLint in the README.

**Comment by jaap3** [comment]
Keeping this open until I'm completely done with the AT follow up, to make sure I don't need to follow up with a fixup PR/release.

**Pull request merged**